### ISSUE TITLE
assert not make sense when release

### ISSUE THREAD

**Issue by lw921014** [NVIDIA/TensorRT#2078]
## Description

In file [https://github.com/NVIDIA/TensorRT/blob/main/samples/common/buffers.h#L251](url),we find this assert in not work when use release build type.
In fact, for the same code, if we compile it using debug mode, code will core dump, however when we compile it using release mode, it works well. I think this is very confused. 

I think the main reason is that in [trt doc](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_network_definition.html)
![image](https://user-images.githubusercontent.com/19748572/174759020-89a68ff8-7bd6-4744-aaba-11a8b364fd52.png)
full dim means all explicit dims, here we have a difference understanding for full dim. Why?

## Environment

**TensorRT Version**: 
**NVIDIA GPU**: 
**NVIDIA Driver Version**: 
**CUDA Version**: 
**CUDNN Version**: 
**Operating System**: 
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by jhalakp-nvidia** [comment]
Are you trying to build an TRT engine using full dims? Full dims means all dimensions are explicit i.e. network does not have implicit batch dimension. 

```
        // Full Dims implies no batch size.
        assert(engine->hasImplicitBatchDimension() || mBatchSize == 0);
```

Above assertion checks if a network with full dims is created, we would expect `mBatchSize==0`. I am not sure this would have a different behavior for release/debug builds.

Can you also specify which version of TensorRT are you using?

**Comment by zerollzeng** [comment]
Closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!
